woodwork/table_accessor.py

```diff
@@ -193,6 +194,10 @@ def __setitem__(self, col_name, column):
         if not isinstance(column, series):
             raise ValueError('New column must be of Series type')
 
+        if column.ww.schema is not None and 'index' in column.ww.semantic_tags:
+            warnings.warn("Cannot allow adding column with index tags. Tag removed", SetItemIndexWarning)
```

Comment:
bit of a nitpick, but `SetItemIndexWarning` makes me think that we're warning users that we are setting the column as an index. Let's call this something like `IndexTagRemovedWarning`.